src/Polly.Shared/RetrySyntaxAsync.cs

```diff
@@ -435,6 +435,46 @@ public static RetryPolicy WaitAndRetryAsync(this PolicyBuilder policyBuilder, in
 
         /// <summary>
         ///     Builds a <see cref="Policy" /> that will wait and retry <paramref name="retryCount" /> times
+        ///     calling <paramref name="onRetry" /> on each retry with the raised exception, the current sleep duration and context data.
```

Comment:
@SteveCote Great stuff navigating all the extra overloads for the async retry syntax.  In each of the four new overloads added to `RetrySyntaxAsync`, we need a minor change to the intellisense documentation here (like you already have in the sync versions):

> on each retry with the raised exception, current sleep duration**, retry count**, and context data.